core/src/main/java/org/jdbi/v3/ReflectiveFieldMapper.java

```diff
@@ -71,33 +72,52 @@ public T map(int row, ResultSet rs, StatementContext ctx)
         for (int i = 1; i <= metadata.getColumnCount(); ++i) {
             String name = metadata.getColumnLabel(i).toLowerCase();
 
-            Field field = properties.get(name);
+            Optional<Field> maybeField = fieldByNameCache.computeIfAbsent(name, this::fieldByColumn);
 
-            if (field != null) {
-                Class type = field.getType();
+            if (!maybeField.isPresent()) {
+                continue;
+            }
 
-                Object value;
-                ResultColumnMapper mapper = ctx.columnMapperFor(type);
-                if (mapper != null) {
-                    value = mapper.mapColumn(rs, i, ctx);
-                }
-                else {
-                    value = rs.getObject(i);
-                }
+            final Field field = maybeField.get();
+            final Class type = field.getType();
+            final Object value;
+            final ResultColumnMapper mapper = ctx.columnMapperFor(type);
 
-                try
-                {
-                    field.setAccessible(true);
-                    field.set(bean, value);
-                }
-                catch (IllegalAccessException e) {
-                    throw new IllegalArgumentException(String.format("Unable to access " +
-                            "property, %s", name), e);
-                }
+            if (mapper != null) {
+                value = mapper.mapColumn(rs, i, ctx);
+            }
+            else {
+                value = rs.getObject(i);
+            }
+
+            try
+            {
+                field.setAccessible(true);
+                field.set(bean, value);
+            }
+            catch (IllegalAccessException e) {
+                throw new IllegalArgumentException(String.format("Unable to access " +
+                        "property, %s", name), e);
             }
         }
 
         return bean;
     }
+
+    private Optional<Field> fieldByColumn(String columnName)
+    {
+        Class<?> aClass = type;
+        while(aClass != null) {
+            for (Field field : aClass.getDeclaredFields()) {
+                for (ColumnNameMappingStrategy strategy : nameMappingStrategies) {
+                    if (strategy.nameMatches(field.getName(), columnName)) {
+                        return Optional.of(field);
+                    }
+                }
+            }
+            aClass = aClass.getSuperclass();
+        }
+        return Optional.empty();
```

Comment:
I feel like by the end of the first failed lookup, we've already iterated through all possible field/strategy combinations, and we'd get better performance by eagerly caching all the column to field mappings in the constructor.